internal/logger/telemetry_store.go

```diff
@@ -64,6 +64,36 @@ type RuleEvalData struct {
 	// TODO: do we want to store params?
 }
 
+// ProjectTombstone can be used to store project metadata in the context of deletion.
+type ProjectTombstone struct {
+	// Project records the project ID that the request was associated with.
+	Project uuid.UUID `json:"project"`
+
+	// ProfileCount records the number of profiles associated with the project.
+	ProfileCount int `json:"profile_count"`
+	// RepositoriesCount records the number of repositories associated with the project.
+	RepositoriesCount int `json:"repositories_count"`
+	// Entitlements that the projects has.
+	Entitlements []string `json:"entitlements"`
+}
+
+// Equals compares two ProjectTombstone structs for equality.
+func (pt ProjectTombstone) Equals(other ProjectTombstone) bool {
+	if len(pt.Entitlements) != len(other.Entitlements) {
+		return false
+	}
+
+	for i, v := range pt.Entitlements {
+		if v != other.Entitlements[i] {
+			return false
+		}
+	}
```

Comment:
Ooops, I just realized that there is [`slices.Equal`](https://pkg.go.dev/golang.org/x/exp/slices#Equal) that should do all this.